### PR TITLE
fix(clerk-js): Hide signup input form when an oauth exists but we have no auth factor

### PR DESCRIPTION
**Pull request by npetridis**
## Type of change

- [X] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [X] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

Hide signup input form when an oauth exists but we have no auth factor
https://www.notion.so/clerkdev/Hide-name-in-SignUp-when-OAuth-is-the-only-factor-dd5341fba63c4237a7f66672d234e97a

### CODE REVIEW

File: packages/clerk-js/src/ui/signUp/SignUpStart.tsx [hunk 1]
```diff
@@ -302,7 +302,12 @@ function _SignUpStart(): JSX.Element {
       </Control>
     ) : null;
 
-  const atLeastOneFormField = nameField || usernameField || emailAddressField || phoneNumberField || passwordField;
+  const atLeastOneAuthFactor =
+    userSettings.attributes.email_address.used_for_first_factor ||
+    userSettings.attributes.phone_number.used_for_first_factor ||
+    (userSettings.attributes.password.required && userSettings.attributes.username.required);
+
+  const showFormFields = atLeastOneAuthFactor || !oauthOptions.length;
```
Comment (SokratisVidros) [inline packages/clerk-js/src/ui/signUp/SignUpStart.tsx:310]: 🔧 @npetridis we should add another check for Web3 options. They behave in the exact same way as oauth options.

File: packages/clerk-js/src/ui/signUp/SignUpStart.tsx [hunk 2]
```diff
@@ -302,7 +302,12 @@ function _SignUpStart(): JSX.Element {
       </Control>
     ) : null;
 
-  const atLeastOneFormField = nameField || usernameField || emailAddressField || phoneNumberField || passwordField;
+  const atLeastOneAuthFactor =
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/signUp/SignUpStart.tsx:305]: ❓ Would it make sense to extract this as a `hasValidAuthFactor` method on the `UserSettings` class?

Eg: 
```tsx
  get hasValidAuthFactor() {
    return this.attributes.email_address.used_for_first_factor ||
    this.attributes.phone_number.used_for_first_factor ||
    (this.attributes.password.required && this.attributes.username.required);
  }

```

File: packages/clerk-js/src/ui/signUp/SignUpStart.tsx [hunk 3]
```diff
@@ -302,7 +302,7 @@ function _SignUpStart(): JSX.Element {
       </Control>
     ) : null;
 
-  const atLeastOneFormField = nameField || usernameField || emailAddressField || phoneNumberField || passwordField;
+  const showFormFields = userSettings.hasValidAuthFactor || (!oauthOptions.length && !oauthOptions.length);
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/signUp/SignUpStart.tsx:305]: 🔧 
I think this should be checking for web3 options, the `(!oauthOptions.length && !oauthOptions.length)` check is probably wrong

**Review by nikosdouvlis — CHANGES_REQUESTED** [review]

File: packages/clerk-js/src/ui/signUp/SignUpStart.tsx (hunk 3 above)
Comment (npetridis) [reply packages/clerk-js/src/ui/signUp/SignUpStart.tsx]: wrong copy/paste when backtesting the fix :P

**Review by nikosdouvlis — APPROVED** [review]

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**